jib-core/src/test/java/com/google/cloud/tools/jib/http/TestWebServer.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.http;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Simple local web server for testing. */
+public class TestWebServer implements Closeable {
+
+  private final ServerSocket serverSocket;
+  private final List<Socket> sockets = new ArrayList<>();
+
+  public TestWebServer() throws IOException, InterruptedException {
+    serverSocket = new ServerSocket(0);
+    new Thread(this::serve200).start();
```

Comment:
An alternative here would be to do `Executors.newSingleThreadExecutor().submit(this::serve200);` and change `serve200` to `private Void serve200() throws IOException` to remove the empty `catch`.